### PR TITLE
Fix auto salvage replicas addressing remount logic

### PR DESCRIPTION
**Pull request by c3y1huang**
The handlePodDeletionIfVolumeRequestRemount only handles pod with controller. And the livenessProbe only restarts the container inside of the pod, therefore volume doesn't get re-mounted as expected. Use StatefulSet to fix test failures caused by "pod with crashed replicas never gets deleted for auto salvage".

Also, removed rebuild since it's no longer required with reusable replicas.

https://github.com/longhorn/longhorn/issues/1914

### CODE REVIEW

**Review by PhanLe1010 — COMMENTED** [review]
In general LGTM. Left some comments.

File: manager/integration/tests/common.py [hunk 1]
```diff
@@ -1659,6 +1659,11 @@ def pvc_name(request):
     return generate_volume_name()
 
 
+@pytest.fixture
+def sts_name(request):
+    return generate_volume_name()
```
[the remark this answers is not in the document]
Comment (c3y1huang) [reply manager/integration/tests/common.py]: Addressed.

File: manager/integration/tests/common.py [hunk 2]
```diff
@@ -3513,6 +3518,57 @@ def expand_attached_volume(client, volume_name):
...
+                   info['pod_name'], 'default', command=["sync"],
+                   stderr=True, stdin=False, stdout=True, tty=False)
+
+            vol_name = v.name
+            pod_name = info['pod_name']
+            break
+
+    assert vol_name is not None
+    assert pod_name is not None
+    assert md5sum is not None
```
[the remark this answers is not in the document]
Comment (c3y1huang) [reply manager/integration/tests/common.py:3562]: Addressed.

File: manager/integration/tests/common.py [hunk 3]
```diff
@@ -3513,6 +3518,57 @@ def expand_attached_volume(client, volume_name):
...
+            'name': 'pod-data'
+        },
+        'spec': {
+            'accessModes': [
+                'ReadWriteOnce'
+            ],
+            'storageClassName': DEFAULT_STORAGECLASS_NAME,
+            'resources': {
+                'requests': {
+                    'storage': data_size_in_mb
```
[the remark this answers is not in the document]
Comment (c3y1huang) [reply manager/integration/tests/common.py]: Addressed.

**Review by PhanLe1010 — APPROVED** [review]
LGTM. 
Just make sure to re-run and re-verify the test case before merging

**Comment by yasker** [comment]
I assume this test has been run and verified.

**Pull request merged**